### PR TITLE
Add context_logits for eval accuracy calculation in case of multi token prediction tasks

### PR DESCRIPTION
**Pull request by athitten**
# What does this PR do ?

This PR adds the following changes:

1) Uses `context_logits` to compute `logProbs` in case of eval benchmarks that have multiple token prediction. For ex: `arc_challenge`, `arc_easy`, `winogrande`, `copa` etc., (where as `MMLU`, `lambada` are single token prediction benchmarks. Uses `generation_logits` for these benchmarks to avoid large payload from `context_logits` )

2) In order to get `context_logits` for evals, the PR exposes `gather_context_logits` and `output_context` logits in `export` and `deploy` files similar to previously existed `generation_logits`

3) Introduces `requirements_eval.txt` file to install `lm-eval-harness` in NeMo FW containers. 

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by oyilmaz-nvidia — COMMENTED** [review]
Looks good but can you please run the examples here https://docs.nvidia.com/nemo-framework/user-guide/latest/deployment/llm/optimized/tensorrt_llm.html and make sure nothing is broken?

**Review by hemildesai — APPROVED** [review]

**Comment by hemildesai** [comment]
LGTM for `nemo/collections/llm` related changes.

**Comment by athitten** [comment]
> Looks good but can you please run the examples here https://docs.nvidia.com/nemo-framework/user-guide/latest/deployment/llm/optimized/tensorrt_llm.html and make sure nothing is broken?

Hi @oyilmaz-nvidia ran the scripts you pointed with HF llama3-8b converted to nemo2. Here's the output

``` Python
python scripts/deploy/nlp/deploy_triton.py --nemo_check
point /workspace/hf_llama3_8b_nemo2_new.nemo --model_type 'llama' --triton_model_name 'llama3-8b' -
-tensor_parallelism_size 1
```
Output of `deploy_triton.py`

``` Python
| cuda_memory_pool_byte_size{1}    | 67108864                                 |
| min_supported_compute_capability | 6.0                                      |
| strict_readiness                 | 1                                        |
| exit_timeout                     | 30                                       |
| cache_enabled                    | 0                                        |
+----------------------------------+------------------------------------------+

I0110 00:42:14.989638 1098739 grpc_server.cc:2558] "Started GRPCInferenceService at 0.0.0.0:8001"
I0110 00:42:14.989788 1098739 http_server.cc:4713] "Started HTTPService at 0.0.0.0:8000"
I0110 00:42:15.030635 1098739 http_server.cc:362] "Started Metrics Service at 0.0.0.0:8002"
E0110 00:42:15.571184 1098739 model_repository_manager.cc:470] "Failed to set config modification time: model_config_content_name_ is empty"
I0110 00:42:15.571553 1098739 model_lifecycle.cc:472] "loading: llama3-8b:1"
I0110 00:42:17.003485 1098739 python_be.cc:2249] "TRITONBACKEND_ModelInstanceInitialize: llama3-8b_0_0 (CPU device 0)"
I0110 00:42:17.484886 1098739 model_lifecycle.cc:839] "successfully loaded 'llama3-8b'"
[01/10/2025-00:42:17] Model serving on Triton is will be started.
```
Inference request:

python scripts/deploy/nlp/query.py -mn 'llama3-8b' -p "Hi, how are you?" -mol 20

Output:
Screen Shot 2025-01-09 at 4 48 53 PM

**Review by oyilmaz-nvidia — APPROVED** [review]
LGTM.

**Review by oyilmaz-nvidia — APPROVED** [review]
